src/main/webapp/app/entities/programming-exercise/form/programming-exercise-due-date-select.component.ts

```diff
@@ -86,14 +86,22 @@ export class ProgrammingExerciseDueDateSelectComponent implements OnChanges {
      * Set the due date. When the due date is set it needs to be checked if the automatic submission date is also set - this should then be updated, too.
      * @param dueDate of the programming exercise.
      */
-    public updateDueDate(dueDate: string) {
-        const updatedDueDate = moment(dueDate).isValid() ? moment(dueDate) : null;
-        const updatedProgrammingExercise = { ...this.exercise, dueDate: updatedDueDate && updatedDueDate.isValid() ? updatedDueDate : null };
+    public updateDueDate(dueDate: moment.Moment | null) {
+        const updatedProgrammingExercise = {
+            ...this.exercise,
+            dueDate: dueDate && dueDate.isValid() ? dueDate : null,
+            buildAndTestStudentSubmissionsAfterDueDate: dueDate ? this.exercise.buildAndTestStudentSubmissionsAfterDueDate : null,
+        };
+        if (!dueDate) {
+            this.buildAndTestDateActive = false;
+        }
         this.buildAndTestDateInvalid =
-            !updatedDueDate ||
-            !updatedDueDate.isValid() ||
+            (this.exercise.buildAndTestStudentSubmissionsAfterDueDate && (!dueDate || !dueDate.isValid())) ||
```

Comment:
This expression is very hard to understand, maybe you can split it or add a comment to better understand how it works. Also I see that you are checking `this.exercise.buildAndTestStudentSubmissionsAfterDueDate` twice, but one time with double negation and another time withour double negation. Is it intended?

Reply:
You are right, I noticed this now, too, when reading through it again. I created single variables to make it easier to read.